### PR TITLE
feat: add BitVec.toInt_[pos|neg]_iff

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
Feedback on proper naming is much appreciated.

**Comment by tobiasgrosser** [comment]
awaiting-review

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -303,6 +303,14 @@ theorem toInt_ofNat {n : Nat} (x : Nat) :
 @[simp] theorem ofInt_natCast (w n : Nat) :
   BitVec.ofInt w (n : Int) = BitVec.ofNat w n := rfl
 
+theorem toInt_neg_iff_toNat_large {w : Nat} {x : BitVec w} :
+    BitVec.toInt x < 0 ↔ 2 ^ w ≤ 2 * x.toNat := by
+  simp [toInt_eq_toNat_cond]; omega
+
+theorem toInt_pos_iff_toNat_small {w : Nat} {x : BitVec w} :
+    0 ≤ BitVec.toInt x ↔ 2 * x.toNat < 2 ^ w := by
+  simp [toInt_eq_toNat_cond]; omega
```
Comment (semorrison) [inline src/Init/Data/BitVec/Lemmas.lean:312]:
```suggestion
theorem toInt_neg_iff {w : Nat} {x : BitVec w} :
    BitVec.toInt x < 0 ↔ 2 ^ w ≤ 2 * x.toNat := by
  simp [toInt_eq_toNat_cond]; omega

theorem toInt_pos_iff {w : Nat} {x : BitVec w} :
    0 ≤ BitVec.toInt x ↔ 2 * x.toNat < 2 ^ w := by
  simp [toInt_eq_toNat_cond]; omega
```

A nice trick with `iff` naming is that if the RHS is sufficiently obvious you don't have to write it out in the name. I'm not sure these really count as obvious, but let's use the trick anyway. :-)

Comment (semorrison) [reply src/Init/Data/BitVec/Lemmas.lean:312]: Please update the PR title too.

Comment (tobiasgrosser) [reply src/Init/Data/BitVec/Lemmas.lean]: Done.

**Pull request merged**